### ISSUE TITLE
[Image Request]: keycloak

### ISSUE THREAD

**Issue by tuananh** [chainguard-images/images#451]
### What software would you like us to add to our image catalog? (one per issue please)

Keycloak https://github.com/keycloak/keycloak

### Add some justification for why this specific image and versions are important.

- Keycloak is a widely popular choice for identity, especially on-premise.
- it's a CNCF project.
- Wolfi already has Keycloak package.

Few thing to note though:

- Keycloak operator (https://github.com/keycloak/keycloak/tree/main/operator) hardcode the `kc.sh` path to `/opt/keycloak/bin/kc.sh` in their code so it's sth we need to keep in mind if we want our image to be compatible with official keycloak operator.

**Comment by dlorenc** [comment]
> ### What software would you like us to add to our image catalog? (one per issue please)
> Keycloak https://github.com/keycloak/keycloak
> 
> ### which versions of the containerized software do you need?
> _No response_
> 
> ### Add some justification for why this specific image and versions are important.
> * Keycloak is a widely popular choice for identity, especially on-premise.
> * it's a CNCF project.
> * Wolfi already has Keycloak package.
> 
> Few thing to note though:
> 
> * Keycloak operator (https://github.com/keycloak/keycloak/tree/main/operator) hardcode the `kc.sh` path to `/opt/keycloak/bin/kc.sh` in their code so it's sth we need to keep in mind if we want our image to be compatible with official keycloak operator.

Good catch! We typically use the /usr/share/java/<app> convention instead of /opt, but this seems like a good enough reason to use /opt or symlink for this specific case.

**Comment by tuananh** [comment]
To be precise, they have a realm import job which uses the same image as the main image.

in that job, they hardcode the path to `kc.sh`: https://github.com/keycloak/keycloak/blob/main/operator/src/main/java/org/keycloak/operator/controllers/KeycloakRealmImportJob.java#L165

```java
        var runBuild = (keycloak.getSpec().getImage() == null) ? "/opt/keycloak/bin/kc.sh build && " : "";

        var commandArgs = List.of("-c",
                runBuild + "/opt/keycloak/bin/kc.sh import --file='" + importMntPath + getRealmName() + "-realm.json' " + override);
```

**Comment by mritunjaysharma394** [comment]
I think these PRs should fix it
* https://github.com/wolfi-dev/os/pull/18780/
* https://github.com/chainguard-images/images/pull/2690
